modules/imgcodecs/src/grfmt_apng.hpp

```diff
@@ -0,0 +1,178 @@
+     * @param width The width of the pixel data.
+     * @param height The height of the pixel data.
+     * @param delayNum The delay numerator for this frame (defaults to DEFAULT_FRAME_NUMERATOR).
+     * @param delayDen The delay denominator for this frame (defaults to DEFAULT_FRAME_DENMINATOR).
+     */
+    APNGFrame(rgba* pixels, int width, int height,
+        int delayNum = DEFAULT_FRAME_NUMERATOR,
+        int delayDen = DEFAULT_FRAME_DENOMINATOR);
+
+    /**
+     * @brief Creates an APNGFrame from a PNG file.
+     * @param filePath The relative or absolute path to an image file.
+     * @param delayNum The delay numerator for this frame (defaults to DEFAULT_FRAME_NUMERATOR).
+     * @param delayDen The delay denominator for this frame (defaults to DEFAULT_FRAME_DENMINATOR).
+     */
+    APNGFrame(const std::string& filePath,
+        int delayNum = DEFAULT_FRAME_NUMERATOR,
+        int delayDen = DEFAULT_FRAME_DENOMINATOR);
+
+    /**
+     * @brief Creates an APNGFrame from a bitmapped array of RBG pixel data.
+     * @param pixels The RGB pixel data.
+     * @param width The width of the pixel data.
+     * @param height The height of the pixel data.
+     * @param delayNum The delay numerator for this frame (defaults to DEFAULT_FRAME_NUMERATOR).
+     * @param delayDen The delay denominator for this frame (defaults to DEFAULT_FRAME_DENMINATOR).
+     */
+    APNGFrame(rgb* pixels, int width, int height,
+        int delayNum = DEFAULT_FRAME_NUMERATOR,
+        int delayDen = DEFAULT_FRAME_DENOMINATOR);
+
+    /**
+     * @brief Creates an APNGFrame from a bitmapped array of RBG pixel data.
+     * @param pixels The RGB pixel data.
+     * @param width The width of the pixel data.
+     * @param height The height of the pixel data.
+     * @param trns_color An array of transparency data.
+     * @param delayNum The delay numerator for this frame (defaults to DEFAULT_FRAME_NUMERATOR).
+     * @param delayDen The delay denominator for this frame (defaults to DEFAULT_FRAME_DENMINATOR).
+     */
+    APNGFrame(rgb* pixels, int width, int height,
+        rgb* trns_color = NULL, int delayNum = DEFAULT_FRAME_NUMERATOR,
+        int delayDen = DEFAULT_FRAME_DENOMINATOR);
+
+    /**
+     * @brief Saves this frame as a single PNG file.
+     * @param outPath The relative or absolute path to save the image file to.
+     * @return Returns true if save was successful.
+     */
+    bool save(const std::string& outPath) const;
+
+    int width() const { return m_width; }
+    int height() const { return m_height; }
+    virtual int type() const { return m_type; }
+
+    // Raw pixel data
+    uchar* pixels(uchar* setPixels = NULL);
+
+    // Palette into
+    rgb* palette(rgb* setPalette = NULL);
+    rgb _palette[256];
+
+    // Transparency info
+    uchar* transparency(uchar* setTransparency = NULL);
+
+    // Sizes for palette and transparency records
+    int paletteSize(int setPaletteSize = 0);
```

Comment:
maybe call the setters: `setPaletteSize`. I like the Goole convention for naming variables: https://google.github.io/styleguide/cppguide.html#Function_Names